app/com/arpnetworking/metrics/portal/query/QueryExecutor.java

```diff
@@ -31,7 +34,6 @@
      *
      * @param query query to execute
      * @return {@link CompletionStage} of the result
-     * @throws QueryExecutionException on errors during query execution
      */
-    CompletionStage<MetricsQueryResult> executeQuery(MetricsQuery query) throws QueryExecutionException;
```

Comment:
I removed this exception from the signature because it didn't make sense to me to have a function return a `CompletableFuture` and also throw an exception.

IMO that error, if it occurs, should happen asynchronously anyway and therefore should be handled via an `exceptionally` combinator from within the `CompetionStage`.